### ISSUE TITLE
Deprecate `@sealed` annotation

### ISSUE THREAD

**Issue by jacob314** [dart-lang/sdk#53310]
The legacy `@sealed` annotation is confusing as its functionality is redundant with the functionality provided by the `final` class modifier.
The one reason people sometimes think they need to use `@sealed` today  instead of `final` today is to have an escape hatch to extend a final class in a different library.
It turns out there is a somewhat esoteric escape hatch that uses the `final` keyword so there is no need for a separate annotation based escape hatch.
https://github.com/dart-lang/language/issues/3106#issuecomment-1593145359
Deprecating the `@sealed` annotation will help push people towards the recommended escape hatch.

**Comment by jacob314** [comment]
Fyi @jakemac53 who had some concerns that we may need to keep sealed or an annotation like `@sealed`. If we were to keep the annotation we should at minimum align the naming scheme and call it `@final` or `@weakFinal` or `@openForTesting`.

**Comment by jakemac53** [comment]
I don't think it is worth renaming, personally. Yes it is more similar to the `final` class modifier in behavior, but it also doesn't perfectly align with that. I would rather either get rid of it or just leave it be 🤷. It is well documented what it does.

**Comment by srawlins** [comment]
`@sealed` means "sealed within the _package_" (or, final within the _package_), so it's not really equivalent to `final` or `sealed`. It's quite a separate beast.

**Comment by srawlins** [comment]
I'm not sure why they would be less desired today than when we wrote the annotation. We initially wanted the annotation to be "library-level" sealing, but there was strong pushback to instead make it package-level. This is still the only annotation that provides that functionality.
